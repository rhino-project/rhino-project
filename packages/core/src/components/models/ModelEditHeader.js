import { breadcrumbFor } from 'rhino/utils/ui';
import { useModelEditContext } from 'rhino/hooks/controllers';
import { useGlobalComponent } from 'rhino/hooks/overrides';
import ModelSection from './ModelSection';

export const ModelEditHeaderBase = (props) => {
  const {
    model,
    show: { resource }
  } = useModelEditContext();

  return (
    <ModelSection baseClassName="edit-header">
      {breadcrumbFor(model, resource, true)}
    </ModelSection>
  );
};

ModelEditHeaderBase.propTypes = {};

const ModelEditHeader = (props) =>
  useGlobalComponent('ModelEditHeader', ModelEditHeaderBase, props);

export default ModelEditHeader;
