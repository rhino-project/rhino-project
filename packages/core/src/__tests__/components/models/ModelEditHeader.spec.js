import { sharedModelTests } from './sharedModelTests';
import ModelEditHeader from 'rhino/components/models/ModelEditHeader';

describe('ModelEditHeader', () => {
  sharedModelTests(ModelEditHeader);
});
