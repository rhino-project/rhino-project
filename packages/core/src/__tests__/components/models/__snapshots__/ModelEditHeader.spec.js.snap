// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModelEditHeader > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelEditHeader > should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
